File: src/com/prey/events/manager/EventManagerRunner.java
package com.prey.events.manager;



import com.prey.PreyLogger;
import com.prey.events.Event;

import android.content.Context;

public class EventManagerRunner implements Runnable{

	private Context ctx=null;
	private Event event;
	
	public EventManagerRunner(Context ctx,Event event){
		this.ctx=ctx;	
		this.event=event;
	}
	
	public void run() {
		if(event!=null){
			PreyLogger.d("CheckInReceiver IN:" + event.getName());
			new EventManager(ctx).execute(event);
			PreyLogger.d("CheckInReceiver OUT:" + event.getName());
		}
	}
}

File: app/src/main/java/com/prey/events/manager/EventManagerRunner.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.events.manager

import android.content.Context

import com.prey.events.Event
import com.prey.PreyLogger

/**
 * EventManagerRunner is a Runnable class responsible for executing events.
 *
 * @param context The application context.
 * @param event The event to be executed.
 */
class EventManagerRunner(var context: Context, var event: Event) : Runnable {

    /**
     * Runs the event execution process.
     */
    override fun run() {
        PreyLogger.d("EVENT CheckInReceiver IN:${event.name}")
        EventManager(context).execute(event)
        PreyLogger.d("EVENT CheckInReceiver OUT:${event.name}")
    }

}
